src/maker/server.rs

```diff
@@ -339,6 +339,12 @@ fn handle_client(
                 }
             }
         }
+
+        // Sometime in macos we get empty messages.
```

Comment:
Can we remove this comment.

Reply:
Why? It's contextual info and describes why we added it. Without it, we can remove the line by mistake.

Reply:
In non-blocking mode, if data is not immediately available, the `read_exact` call will return a `WouldBlock` error instead of waiting for the data to arrive. On Linux, this issue was not observed, but on macOS, there's a subtle delay that more closely resembles how real-world scenarios would behave.

Reply:
We are not setting the stream to non-blocking in this case. The streams are blocking. Each thread will wait for the incoming data. The delay should not be the problem as the thread will wait. 

In any case, blocking or non-blocking, it should not be an empty vec. And as it's still unexplained why that's the case in macos it should be documented.

Reply:
> On Linux, this issue was not observed

The delay is there in Linux too. It simply takes some time for the Taker to process data and send it. So the thread waits.

Reply:
listener thread is non-blocking.
```
 let listener = TcpListener::bind((Ipv4Addr::LOCALHOST, maker.config.port))?;
    log::info!(
        "[{}] Listening for client conns at: {}",
        maker.config.port,
        listener.local_addr()?
    );
    listener.set_nonblocking(true)?;
```

Reply:
This will affect the accept call only. Not the read calls. That means if no incoming connection is received it would not block. That's not the same as `Tcpstream::set_nonblocking()`.

Reply:
When listener.set_nonblocking(true) is called, it sets the TcpListener to non-blocking mode. As a result, the accept method on the listener will not block while waiting for incoming connections, it will return immediately and may produce a WouldBlock error if no connection is available. When listener.accept() successfully returns a connection, the resulting TcpStream inherits this non-blocking property from the listener, making the stream non-blocking by default. In a non-blocking TcpStream, read and write operations do not wait for data, they immediately return with a WouldBlock error if no data is available to read or write. This stream is then passed to the handle_client function, which runs on a separate thread but continues to use the same non-blocking TcpStream. Given this, also the set_timeout settings for read and write operations on the stream are redundant, as they do not affect the non-blocking behavior.

Reply:
> When listener.accept() successfully returns a connection, the resulting TcpStream inherits this non-blocking property from the listener,

Where are you getting this from? That doesn't seem to be the case. Or else we would have to handle the Wouldblocks. We don't.